server/src/main/java/org/opensearch/action/search/SearchShardTask.java

```diff
@@ -84,4 +86,12 @@ public boolean supportsResourceTracking() {
     public boolean shouldCancelChildrenOnCancellation() {
         return false;
     }
+
+    public String getResourceLimitGroupName() {
+        return resourceLimitGroupId;
+    }
+
+    public void setResourceLimitGroupName(String resourceLimitGroupId) {
+        this.resourceLimitGroupId = resourceLimitGroupId;
+    }
```

Comment:
The reason to have this set using setter is that the Task is created way up in the stack: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/TransportAction.java#L101
So we won't have the **resourceLiimitGroupId** but we will change the information is coming in the searchRequest so this will likely not be present in final change.